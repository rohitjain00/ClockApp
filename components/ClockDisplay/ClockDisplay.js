import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import Moment from 'moment';
import PropTypes from 'prop-types';

export function ClockDisplay(props) {
    function dateStyle() {
        return {
            fontSize: props.dateFontSize,
            fontFamily: props.dateFontFamily,
            color: props.dateFontColor,
        };
    }

    function timeStyle() {
        return {
            fontSize: props.timeFontSize,
            fontFamily: props.timeFontFamily,
            color: props.timeFontColor,
        };
    }

    return (
        <View style={styles.container}>
            <Text style={timeStyle()}>{Moment(props.dateTime).utcOffset(props.zone).format(props.timeFormat)}</Text>
            <Text style={dateStyle()}>{Moment(props.dateTime).utcOffset(props.zone).format(props.dateFormat)}</Text>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
        // marginTop: "100px"
    },
});

ClockDisplay.propTypes = {
    dateFontSize: PropTypes.number,
    dateFontFamily: PropTypes.string,
    dateFontColor: PropTypes.string,
    timeFontSize: PropTypes.number,
    timeFontFamily: PropTypes.string,
    timeFontColor: PropTypes.string,
    timeFormat: PropTypes.string,
    dateFormat: PropTypes.string,
    dateTime: PropTypes.instanceOf(Date),
    zone: PropTypes.number
}