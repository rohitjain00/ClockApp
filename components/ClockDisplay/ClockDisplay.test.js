//ClockDisplay.test.js
import React from 'react';
import {render} from '@testing-library/react-native';
import { ClockDisplay } from "./ClockDisplay";
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ClockDisplay Component', () => {
    const date = new Date(2020, 7, 30, 10, 10, 10);
    const component = <ClockDisplay  dateTime={date} timeFontSize={20} dateFontSize={20} dateFormat={'DD/MM/YYYY'} timeFormat={'HH:mm:ss A'} dateFontFamily={'normal'} timeFontFamily={'normal'} dateFontColor={'#000000'} timeFontColor={'#000000'} zone={0} />
    it('Clock Display Props Validation', async () => {
        const wrapper = mount(component);
        const componentInstance = wrapper.props();
        expect(componentInstance.timeFontSize).toEqual(20);
        expect(componentInstance.dateFontSize).toEqual(20);
        expect(componentInstance.dateFormat).toEqual('DD/MM/YYYY');
        expect(componentInstance.timeFormat).toEqual('HH:mm:ss A');
        expect(componentInstance.dateFontFamily).toEqual('normal');
        expect(componentInstance.timeFontFamily).toEqual('normal');
        expect(componentInstance.dateFontColor).toEqual('#000000');
        expect(componentInstance.timeFontColor).toEqual('#000000');
        expect(componentInstance.zone).toEqual(0);
    });
    it('Clock Display Snapshot Validation', async () => {
        const tree = render(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});