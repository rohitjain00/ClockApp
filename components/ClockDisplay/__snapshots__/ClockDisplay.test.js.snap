// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClockDisplay Component Clock Display Snapshot Validation 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "normal",
        "fontSize": 20,
      }
    }
  >
    04:40:10 AM
  </Text>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "normal",
        "fontSize": 20,
      }
    }
  >
    30/08/2020
  </Text>
</View>
`;
